lib/libvfio-user.c

```diff
@@ -1376,6 +1396,9 @@ vfu_run_ctx(vfu_ctx_t *vfu_ctx)
     blocking = !(vfu_ctx->flags & LIBVFIO_USER_FLAG_ATTACH_NB);
 
     do {
+        if (vfu_ctx->migr_trans_pending) {
```

Comment:
I think we need the same for at least vfu_dma_read/write()
For vfu_irq_trigger() we can (probably?) get away with it for now, until it needs the socket.

Reply:
I've documented that it isn't allowed so I'll only add an `assert`.